lib/AST/ASTVerifier.cpp

```diff
@@ -3034,8 +3034,11 @@ class Verifier : public ASTWalker {
       PrettyStackTraceDecl debugStack("verifying DestructorDecl", DD);
 
       auto *ND = DD->getDeclContext()->getSelfNominalTypeDecl();
-      if (!isa<ClassDecl>(ND) && !DD->isInvalid()) {
-        Out << "DestructorDecls outside classes should be marked invalid";
+      // Destructors are allowed in all classes and in structs imported from
+      // C++.
+      if (!isa<ClassDecl>(ND) && !DD->isInvalid() && !ND->getClangDecl()) {
```

Comment:
Is this still needed? I see in `VisitCXXDestructorDecl` that you're never actually generating a Swift deinitializer now?

I think it would be better if this wasn't needed -- in effect, this is an extension to the Swift language, as the deinitializer will be visible on the struct's API (I believe?), and that will seem strange to people looking at the API.

Reply:
Yep, you're right. Removed.